src/read/reader.rs

```diff
@@ -31,6 +31,7 @@ fn is_primitive(data_type: &DataType) -> bool {
 pub struct NativeReader<R: NativeReadBuf> {
     page_readers: Vec<R>,
     field: Field,
+    is_nested: bool,
```

Comment:
is_nested  can be inferred from field, so it's not necessary in this constructor.

Reply:
For tuple data type, we may read the inner columns directly, in this case `NativeReader` don't know that the column is a inner column of the tuple.